extensions/amp-inputmask/validator-amp-inputmask.protoascii

```diff
@@ -27,30 +27,118 @@ tags: {  # amp-inputmask
   }
   attr_lists: "common-extension-attrs"
 }
+
 # The input element is also defined by validator/validator-main.protoascii
 tags: {
   html_format: AMP
   tag_name: "INPUT"
-  spec_name: "input [mask]"
+  spec_name: "input [mask] (custom mask)"
   requires_extension: "amp-inputmask"
   attrs: {
     name: "mask"
     mandatory: true
     dispatch_key: NAME_DISPATCH
```

Comment:
Nevermind. I looked at the code and I'm wrong. More specific dispatch keys take precedence over less specific, so as long as you don't have two keys with the exact same rule, which you don't, this will work. We'll match against the NAME_VALUE dispatch keys first, followed by just NAME

Reply:
Thanks for taking a look @Gregable! The tests in [extensions/amp-inputmask/0.1/test/validator-amp-inputmask.html](https://github.com/cvializ/amphtml/blob/20be4d4f118a733141ef406c328d3d6c5b9a6309/extensions/amp-inputmask/0.1/test/validator-amp-inputmask.html) give an overview of what I'm trying to make valid and invalid.

For clarity, the goal is to allow arbitrary custom masks and a set of named masks, but if a named mask is used it should be the only mask in the attribute.